hathor_explorer_v1.md

```diff
@@ -0,0 +1,766 @@
+Network data will also be improved, showing the overall network instead of a single node point of view.
+
+> **Estimated cost:** $1.50 monthly ($ 0.50 per million update, every second update).
+
+### Statistics Collector
+[guide-level-explanation/hathor-explorer-service/statistics-collector]: #guide-level-explanation/hathor-explorer-service/statistics-collector
+
+[`data-collector`](#guide-level-explanation/hathor-explorer-service/data-collector) will listen to hashrate/statistics updates and send to a SNS. A SQS will listen to this SNS and store it.
+Statistics collector updater will consume this SQS and store data in the database.
+Statistics collector will also trigger [`statistics-update-handler`](#guide-level-explanation/hathor-explorer-service/statistics-update-handler) to update connected clients. 
+
+> **Estimated cost:** $ 0.90 per million transactions.
+
+### Data Collector
+[guide-level-explanation/hathor-explorer-service/data-collector]: #guide-level-explanation/hathor-explorer-service/data-collector
+
+Data Collector will connect to `full-node` WebSockets to listen for updates and send messages to SNS topics.
+For data which doesn't come from WebSocket, polling will be used. 
+
+> **Estimated cost:** $ 14 monthly (`t3.small`).
+### API gateway
+[guide-level-explanation/hathor-explorer-service/api-gateway]: #guide-level-explanation/hathor-explorer-service/api-gateway
+
+API gateway will handle REST requests and WS connections from the external world. Also will cache requests strategically to prevent services overload.
+
+> **Estimated cost:** $2 monthly (according to AWS example. Variables are quantity of requests, time of connection and data transferred).
+## Hathor Explorer Lib
+[guide-level-explanation/hathor-explorer-lib]: #guide-level-explanation/hathor-explorer-lib
+
+As is a good practice use React components as presentation layer only, Hathor Explorer Lib comes as an abstraction for business matters. All requests, WS connections, Entities and business decisions will be made by it and imported by [`hathor-explorer`](#guide-level-explanation/hathor-explorer) to be called.
+
+It's a good idea to separate things inside the project first and extract to a different one later. So it will still be part of [`hathor-explorer`](#guide-level-explanation/hathor-explorer) for now.
+
+## Hathor Explorer
+[guide-level-explanation/hathor-explorer]: #guide-level-explanation/hathor-explorer
+
+The existing project will not have any big visual modification except by network visualization, token list pagination and the hashrate x time graph that will be "all time" instead of starting only by the time the page is loaded.
+
+Furthermore business logic will be extracted from React components and API calls will be modified, if needed, to fit the new responses from new [guide-level-explanation/hathor-explorer-service](#guide-level-explanation/hathor-explorer-service)
+
+> **Estimated total monthly cost:** $ 56.00 with a huge margin of accesses
```

Comment:
What's this cost based on? What kinds of AWS services are considered in this?

Reply:
@luislhl each section above have the estimated cost. This is the sum of all. I'm using AWS "pricing" pages as reference.